src/TgSharp.Core/Network/MtProtoPlainSender.cs

```diff
@@ -7,21 +7,37 @@ namespace TgSharp.Core.Network
 {
     public class MtProtoPlainSender
     {
+        public Action<byte[]> OnResponseReceived;
+
         private int timeOffset;
         private long lastMessageId;
         private Random random;
-        private TcpTransport transport;
+        private WSTransport transport;
 
-        public MtProtoPlainSender(TcpTransport transport)
+        public MtProtoPlainSender(WSTransport transport)
         {
             this.transport = transport;
+            transport.OnUnencryptedMessage += Transport_OnUnencryptedMessage;
```

Comment:
do we need to unsuscribe this at dispose/reconnection time? otherwise there's a leak

Reply:
ping ^